### PR TITLE
Add option to filter board cards by labels and assignees

### LINKED ISSUE #21846: Project board card filtering
### Feature Description

As a developer of a large software project I’d like to be able to filter project boards to see the status of tickets that are relevant to my contribution area. A real life example would be: If we had a project board for a specific Gitea release, and my area of focus was the project board feature, I could use this filtering to track the status of tickets for the project board with still being able to have an overall board that a release manager would be able to have a fuller view.

### Screenshots

Mockups of what this might look like are:
board1
board2

By default all issues would appear on the project board, however if you’d like to only have issues with a specific label (or multiple labels) you could either click on the label attached to the card, or use the multi-select dropdown. Once a label (or multiple labels) has been selected, then only the issues with those labels would show (as seen in the second screenshot).

In Gitea currently, this filtering exists on the issue/pr list pages, and generates a URL that can be shared to show others the filtered view (or bookmarked for quick access).

Other implementations:
In GitHub's project boards, they have a search box that allows for the same filtering that they have on the issues/pr list.
image
image

You can read more about their filters here:                    
https://docs.github.com/en/github-ae@latest/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards/filtering-cards-on-a-project-board

JIRA also has filtering based on a query language for their project boards

### Implementation Details

The filtering for issues would be in the `ViewProject` function (https://github.com/go-gitea/gitea/blob/main/routers/web/repo/projects.go#L275), and the filtering concept could be taken from https://github.com/go-gitea/gitea/blob/main/routers/web/repo/issue.go#L263

There is a difficulty though, when items on the board are moved, they send a request to the "move" endpoint, and that request contains the sort and location of each issue in the board, but if the server side filtering is applied, then the issues that are filtered out wouldn't be included. A quick/naive way to address this, would be to apply the filtering client side.

### PR DESCRIPTION
**Pull request by lafriks**
Works in both organization and repository project boards

Fixes #21846

Replaces #21963
Replaces #27117
 
image

**Note** that implementation was made intentionally to work same as in issue list so that URL can be bookmarked for quick access with predefined filters in URL

### CODE REVIEW

**Comment by techknowlogick** [comment]
Thanks @lafriks!! <3

I haven't tested this yet, but how does it handle the case with changing sort when some items are hidden?

**Comment by lunny** [comment]
How did you load the labels for the repository project, org project, and user projects. I think all of them are different.
- For repository project, we just need to load the labels of this repository. 
- For org project, we may need to load all the labels of all repositories of this org as well as the org' labels.
- For user project, since there is no user level labels, we need to load all the labels of all repositories of this user.

**Comment by lafriks** [comment]
For labels I did:

* Repo project: Load org (if organization is owner) and repo labels (same as issue list is doing)

* Org projects: load only organization labels as adding labels for all repositories would be overkill imho and would be hard to merge same named labels from different repos together

For assignees I did:

* Repo project: all assignable users for that repo

* Org projects: all assignable users for all that organization repos

**Comment by lafriks** [comment]
> I haven't tested this yet, but how does it handle the case with changing sort when some items are hidden?

It work okeyish imho, don't see anything broken

[label-filter-sorting.webm](https://github.com/user-attachments/assets/cc933e21-9b4a-488c-88cc-64285a4f2b77)

**Comment by lunny** [comment]
> * Org projects: load only organization labels as adding labels for all repositories would be overkill imho and would be hard to merge same named labels from different repos together

Looks like for Org projects, since the issues on this project may also contain repository labels, how did we handle the situation?

**Comment by lafriks** [comment]
Currently in org projects you can not filter by repo only labels but that imho can be left for future improvement, mostly if you are using org projects you should be using org labels 😅

**Comment by lunny** [comment]
> Currently in org projects you can not filter by repo only labels but that imho can be left for future improvement, mostly if you are using org projects you should be using org labels 😅

OK. Please add some TODO or FIXME there.

File: routers/web/org/projects.go [hunk 1]
```diff
@@ -372,6 +395,45 @@ func ViewProject(ctx *context.Context) {
 		}
 	}
 
+	labels, err := issues_model.GetLabelsByOrgID(ctx, project.OwnerID, "", db.ListOptions{})
```
Comment (lafriks) [inline routers/web/org/projects.go:399]:
```suggestion
	// TODO: Add option to filter also by repository specific labels
	labels, err := issues_model.GetLabelsByOrgID(ctx, project.OwnerID, "", db.ListOptions{})
```

**Comment by lafriks** [comment]
@lunny added todo comment

**Review by lunny — APPROVED** [review]

**Comment by techknowlogick** [comment]
@lafriks amazing! Thank you so much!

**Review by techknowlogick — APPROVED** [review]

File: models/organization/org_user.go [hunk 2]
```diff
@@ -112,6 +114,49 @@ func IsUserOrgOwner(ctx context.Context, users user_model.UserList, orgID int64)
 	return results
 }
 
+// GetOrgAssignees returns all users that have write access and can be assigned to issues
+// of the any repository in the organization.
+func GetOrgAssignees(ctx context.Context, orgID int64) (_ []*user_model.User, err error) {
```
Comment (wxiaoguang) [inline models/organization/org_user.go:119]: This logic seems not right.

-> Fix org team assignee/reviewer lookups for team member permissions #37365

**Pull request merged**